spyder/plugins/console/widgets/main_widget.py

```diff
@@ -0,0 +1,535 @@
+        self.add_item_to_menu(
+            run_action,
+            menu=options_menu,
+            section=ConsoleWidgetOptionsMenuSections.Run,
+        )
+        self.add_item_to_menu(
+            environ_action,
+            menu=options_menu,
+            section=ConsoleWidgetOptionsMenuSections.Run,
+        )
+        self.add_item_to_menu(
+            syspath_action,
+            menu=options_menu,
+            section=ConsoleWidgetOptionsMenuSections.Run,
+        )
+        self.add_item_to_menu(
+            internal_settings_menu,
+            menu=options_menu,
+            section=ConsoleWidgetOptionsMenuSections.Run,
+        )
+        self.add_item_to_menu(
+            quit_action,
+            menu=options_menu,
+            section=ConsoleWidgetOptionsMenuSections.Quit,
+        )
+
+        self.shell.set_external_editor(
+            self.get_option('external_editor/path'), '')
+        self.start_interpreter(self.get_option('namespace'))
+
+    def on_option_update(self, option, value):
+        if option == 'max_line_count':
+            self.shell.setMaximumBlockCount(value)
+        elif option == 'wrap':
+            self.shell.toggle_wrap_mode(value)
+        elif option == 'codecompletion/auto':
+            self.shell.set_codecompletion_auto(value)
+        elif option == 'external_editor/path':
+            self.shell.set_external_editor(value, '')
+
+    def update_actions(self):
+        pass
+
+    def get_focus_widget(self):
+        return self.shell
+
+    # --- Qt overrides
+    # ------------------------------------------------------------------------
+    def dragEnterEvent(self, event):
+        """
+        Reimplement Qt method.
+
+        Inform Qt about the types of data that the widget accepts.
+        """
+        source = event.mimeData()
+        if source.hasUrls():
+            if mimedata2url(source):
+                event.acceptProposedAction()
+            else:
+                event.ignore()
+        elif source.hasText():
+            event.acceptProposedAction()
+
+    def dropEvent(self, event):
+        """
+        Reimplement Qt method.
+
+        Unpack dropped data and handle it.
+        """
+        source = event.mimeData()
+        if source.hasUrls():
+            pathlist = mimedata2url(source)
+            self.shell.drop_pathlist(pathlist)
+        elif source.hasText():
+            lines = to_text_string(source.text())
+            self.shell.set_cursor_position('eof')
+            self.shell.execute_lines(lines)
+
+        event.acceptProposedAction()
+
+    # --- API
+    # ------------------------------------------------------------------------
+    def start_interpreter(self, namespace):
+        """
+        Start internal console interpreter.
+        """
+        self.shell.start_interpreter(namespace)
+
+    def set_historylog(self, historylog):
+        """
+        Bind historylog instance to this console.
+
+        Not used anymore since v2.0.

```

Comment:
Please remove this line because I think it's inaccurate.

Reply:
Nope, it is actually accurate. The history plugin does not use this since a long time. 

But maybe it is a bit of a misleading comment.